src/control/lib/netdetect/netdetect.go

```diff
@@ -1059,19 +1059,17 @@ func ValidateNUMAConfig(device string, numaNode uint) error {
 		return err
 	}
 	if deviceAffinity.NUMANode != numaNode {
-		return errors.Errorf("The NUMA node for device %s does not match the provided value %d. "+
-			"Remove the pinned_numa_node value from daos_server.yml then execute 'daos_server network scan' "+
-			"to see the valid NUMA node associated with the network device", device, numaNode)
+		return errors.Errorf("The NUMA node for device %s does not match the provided value %d.", device, numaNode)
 	}
 	log.Debugf("The NUMA node for device %s matches the provided value %d.  Network configuration is valid.", device, numaNode)
 	return nil
 }
 
-func createFabricScanEntry(deviceScanCfg DeviceScan, provider string, devCount int, resultsMap map[string]struct{}, ScanResults []FabricScan, excludeMap map[string]struct{}) (map[string]struct{}, []FabricScan, error) {
+func createFabricScanEntry(deviceScanCfg DeviceScan, provider string, devCount int, resultsMap map[string]struct{}, ScanResults []FabricScan, excludeMap map[string]struct{}) (FabricScan, error) {
 	log.Debugf("Device scan target device name: %s", deviceScanCfg.targetDevice)
 	deviceAffinity, err := GetAffinityForDevice(deviceScanCfg)
 	if err != nil {
-		return resultsMap, ScanResults, err
+		return FabricScan{}, err
```

Comment:
If your function signature specified that the return type is `*FabricScan` instead, you could just return nil in the error case, which is idiomatic for Go.

Reply:
fixed